.github/workflows/tests.yaml

```
@@ -59,15 +31,69 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install Base Dependencies
         run: |
+          echo $COVERAGE_FILE
           python -m pip install --upgrade pip setuptools wheel
           pip install --upgrade tox
-      - name: Run Tests (Tox)
-        run: tox -e py
-      - uses: 5monkeys/cobertura-action@master
-        name: Upload Coverage Report
+      - name: Run ${{ matrix.tests }} Tests
+        run: |
+          echo $COVERAGE_FILE
+          tox -e ${{ matrix.tests }}
+          ls -al
+          ls -al .tox
+      - uses: actions/upload-artifact@v2
         if: ${{ matrix.python-version == 3.9}}
         with:
-          path: coverage.xml
-          repo_token: ${{ secrets.GITHUB_TOKEN }}
-          minimum_coverage: 75
-          fail_below_threshold: false
+          name: coverage {{ matrix.tests }}
+          path: ${{ env.COVERAGE_FILE }}
```

Is this needed because of the random ids appended by `coverage run -p`?
```suggestion
          path: ${{ env.COVERAGE_FILE }}.*
```